### DISCUSSION TITLE
How to run any .xsh script without preceding xonsh?

### DISCUSSION THREAD

**Discussion by sudg6** [xonsh/xonsh#4475] [Q&A] [answered]
I have Windows and using VS code for editing and its integrated Terminal to run scripts, I want to run a simeple script called Untitled.xsh without invoking xonsh in the front; that is,
```
xonsh Untitled.xsh
```
runs as expected. However, omitting xonsh shows errors,
```
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
xonsh: subprocess mode: command not found: /c/Users/Zidani/AppData/Roaming/Python/Python37/Scripts/xonsh
 ```
seems like Xonsh doesn't support shebang in Windows that I set in Untitled.xsh
```
#!\c\Users\name\AppData\Roaming\Python\Python37\Scripts\xonsh.exe
```

**Comment by sudg6** [comment, marked as answer]
I figured a solution that implies writing the shebang as follows (I already installed Git and enabled Linux commands during Git installation process),
```
#!\C:\\Users\\USER\\AppData\\Roaming\\Python\\Python37\\Scripts\\xonsh
```
As we know shebang is Unix-based syntax but the escape characters should be inverted and doubled because we are using it in the Windows system.
An alternate solution is one should use Git Bash as the default Terminal in VS code then including the shebang command in Untitled.xsh.

**Comment by halloleo** [comment]
I'm not working with xonsh on Windows, but on Unix it works to write just 

    #!xonsh

**when you have xonsh in your path**.

**Reply by agoose77** [reply]
@halloleo this does often work, but it's usually better practice to actually but a full path to the interpreter. In this case, one would use
```xonsh
#!/usr/bin/env xonsh
```
to resolve against `$PATH`

**Reply by halloleo** [reply]
Of course! Thanks for adding this, @agoose77.
